### PR TITLE
Add EUPL v1.2

### PR DESCRIPTION
**Pull request by MattiSG**
- Date of adoption: 18/05/2017.
- Date of effect: 08/06/2017.
- No end.

Mostly updates the list of compatible licenses.

### CODE REVIEW

**Comment by mlinksva** [comment]
Tests are failing because 1.2 does not yet meet the [criteria for cataloging here](https://github.com/github/choosealicense.com/blob/b59fee6b2cf51ab7d137afec12e6d7b9e89692d8/CONTRIBUTING.md#adding-a-license).

I imagine an SPDX ID will be assigned and OSI will formally approve reasonably soon. In the meantime, it would be great to find 3 examples. I'll leave the PR open until it can be merged.

**Comment by MattiSG** [comment]
Thanks for your feedback.

Which 3 examples are you looking for? Software licensed under this license?

**Comment by mlinksva** [comment]
See at the criteria link:

> Identification of 3 notable projects using the license with straightforward LICENSE files which serve as examples newcomers can follow and that could be detected by licensee if it knew about the license

They should be documented in the [using](https://github.com/github/choosealicense.com#required-fields) field.

**Comment by mlinksva** [comment]
Currently only 48 results, but noting https://github.com/search?utf8=%E2%9C%93&q=filename%3ALICENSE+%22EUPL+%C2%A9+the+European+Union+2007%2C+2016%22&ref=simplesearch for help in finding examples later.

**Comment by haphut** [comment]
Digitransit[[1](https://digitransit.fi/en/),[2](https://github.com/HSLdevcom/digitransit)] is a public transit journey planning platform used by public transit authorities in several countries. It's dual-licensed with AGPLv3 and EUPL v1.2. Here's the license file of a particularly important repo: https://github.com/HSLdevcom/digitransit-ui/blob/master/LICENSE-EUPL.txt

**Comment by mlinksva** [comment]
EUPL-1.2 is the id now published by SPDX, so we just need 3 `using:` examples that would be detected. Digitransit above wouldn't be because it's dual licensed.

**Comment by Haroenv** [comment]
There's 116 results on https://github.com/search?utf8=✓&q=filename%3ALICENSE+%22EUPL+©+the+European+Union+2007%2C+2016%22&ref=simplesearch now 🇪🇺

**Comment by MattiSG** [comment]
Thanks @Haroenv, I went through the first pages of results and listed three `using` that are not pet projects 🙂

**Comment by MattiSG** [comment]
The [tests fail](https://travis-ci.org/github/choosealicense.com/builds/324899963#L871-L899) because the [Licensee gem fails to detect](https://github.com/github/choosealicense.com/blob/gh-pages/spec/license_meta_spec.rb#L58) the licenses as `eupl-1.2`… which is a bummer, since `licensee` [depends on](https://github.com/benbalter/licensee/blob/master/docs/CONTRIBUTING.md#adding-a-license) `choosealicense.com` to contain the license in order to detect it 🙃

I thus understand that this is expected behaviour when adding a new license. If this is not the case, guidance on how to fix the tests would be welcome 🙂

File: _licenses/eupl-1.2.txt [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+— Québec Free and Open-Source Licence — Reciprocity (LiLiQ-R) or
+  Strong Reciprocity (LiLiQ-R+)
+
+— The European Commission may update this Appendix to later versions of the
+  above licences without producing a new version of the EUPL, as long as they
+  provide the rights granted in Article 2 of this Licence and protect the
+  covered Source Code from exclusive appropriation.
+— All other changes or additions to this Appendix require the production of a
+  new EUPL version.
+
```
Comment (mlinksva) [inline _licenses/eupl-1.2.txt]: I think everything after this line is a spurious duplication of most of the license text.

**Comment by mlinksva** [comment]
@MattiSG the tests patch licensee to use the local copy of the licenses. I would not expect pi-hole or OpenStaticAnalyzer to match as they each prepend some notices to the license text. AethysRotation probably would match but there's a problem with the license text in this PR, added a review comment on that at https://github.com/github/choosealicense.com/pull/523/files#r159702343

**Comment by MattiSG** [comment]
Indeed, now AethysRotation does pass! Thanks for the pointer!

How do you want to proceed regarding the two other examples? I can I try and identify other projects, but I'm a bit doubtful that's the most pertinent approach considering adding notices is quite a standard practice.

File: _licenses/eupl-1.2.txt (hunk 1 above)
Comment (MattiSG) [reply _licenses/eupl-1.2.txt]: It was indeed an accidental duplication introduced when line wrapping. Fixed.

**Comment by mlinksva** [comment]
We'll likely need other examples.

If the prepended notices are in a standard form conceivably licensee could eventually recognize those, otherwise it won't because they could mean anything https://github.com/benbalter/licensee/issues/239#issuecomment-344350874

**Comment by mlinksva** [comment]
If you or someone else wants to go beyond searching for text in the EUPL-1.2 license, an option would be to see if any [EUPL-1.1 projects](https://github.com/search?q=license%3Aeupl-1.1) want to upgrade. Of course a random request to change licenses can be annoying, so use discretion.

One other note: I want the text here to be the same as published by both the license steward and SPDX (which ought to match anyway). https://spdx.org/licenses/EUPL-1.2.html and if they have anything wrong, https://github.com/spdx/license-list-XML/blob/master/src/EUPL-1.2.xml is the source.

**Comment by MattiSG** [comment]
> I want the text here to be the same

Are you suspecting this is not the case? I copy-pasted, line-wrapped automatically and double-checked the line wrapping. There was a mistake in double-pasting the result in the web editor, but I don't think any of the content was changed.

**Comment by mlinksva** [comment]
I'm not suspecting they're not the same. Thought of it because the text there doesn't include any optional notice header. Apologies for not saying that explicitly.

**Comment by MattiSG** [comment]
Tests failed because https://wiki.creativecommons.org/wiki/CC0_FAQ was down. It now seems to be back online, but I don't have the rights to trigger a new build on Travis, and I already used the “update from master” trick to trigger one. Could an admin do it please? 🙂

**Comment by MattiSG** [comment]
Ok, the checker doesn't follow redirects, or CloudFlare detects it's a bot and doesn't redirect. All builds will fail until https://wiki.creativecommons.org is back online.

**Comment by mlinksva** [comment]
OK, since that's the only test failure and it's known and outside our control, I'm good with merging this. If the unrelated site stays offline we can fix that link later. Thanks for keeping at this for over half a year! :smile:

**Comment by MattiSG** [comment]
🎉 Thanks for your help and speed in replying!

How long do you expect this to take before the EUPL 1.2 is available throughout GitHub (in search filters & license detection)?

I [asked](https://twitter.com/matti_sg/status/949041028039770112) CreativeCommons, we'll see what (if) they answer :wink:

**Comment by mlinksva** [comment]
A new release of licensee has to be cut with updated vendored licenses, like https://github.com/benbalter/licensee/commit/4593ffc585691b9d25c547cf37f332f7126c3963 then deployed to github.com. I don't have an estimate, but you'll be able to tell that it is done by https://api.github.com/licenses/eupl-1.2 giving a result with the content that's been added here.

**Comment by MattiSG** [comment]
Perfect, thanks!

In other news, CreativeCommons [confirms](https://twitter.com/creativecommons/status/949048771169591296) the wiki is back up 👍

**Comment by Eagle3386** [comment]
Is there any (manual) trigger required in order to re-run tests or is the PR accepted anyway?
Additionally, I wrote GH support in order to speed up EUPL-1.2 availability as I explicitly need this license and this explicit version.

As a GH user, I don't see what takes so long and/or work to fix this issue. Therefore, further information about this would be greatly appreciated.

**Comment by mlinksva** [comment]
@Eagle3386 I'm not sure what tests you're referring to. Could you explain more?

I understand that you're eager to have EUPL-1.2 be detected on github.com but just to be clear, you can use the license in your repos now -- just copy the text from https://choosealicense.com/licenses/eupl-1.2/ and paste into a file called `LICENSE` in your repo. The extra functionality on github.com is based off of that simple practice, not the other way around.

**Comment by Eagle3386** [comment]
@mlinksva Sure, I'm referring to those tests mentioned by @MattiSG in his [comment above](523#issuecomment-355411052).

Also, thanks for the hint, but there are two issues with the link you provided:
1. It's not EUPL's original license text. See the [original (English) version](https://joinup.ec.europa.eu/sites/default/files/inline-files/EUPL%20v1_2%20EN(1).txt) which is almost totally different regarding line breaks, additional empty lines, etc.
IMHO, nobody should alter a license's text - the GPL's text remains the same throughout pretty much the whole Internet and that should apply to the EUPL, too.
2. GH is very misleading regarding the `LICENSE` file: some of its help pages suggest the aforementioned name while others encourage to add the file extension `.md` - like `README.md`.

Anyway, my [concerned repo](https://github.com/Eagle3386/aya-platform/) already
- contains a `LICENSE.md` file
- states `licensed under EUPL-1.2` in its description
- includes a `license | EUPL` badge in its `README.md` file.

So, your accusation doesn't seem to be true - I tried as good as I could to trigger GH showing that license badge right next to `Contributors`.
Though, it doesn't and that's why I'm so eager getting this fixed - anyone not willing to accept GPL & Co. is currently just screwed: please see the `Choose a license template` button and its allegedly available licenses at https://github.com/USERNAME/REPO-NAME/community/license/new - you only get a very limited subset of licenses available over at https://choosealicense.com/appendix/:
- Apache License 2
- BSD 2-Clause "Simplified" / 3-Clause "New" or "Revised" License
- Eclipse Public License 2
- GNU Affero GPL v3, GPL v2/3, LGPL v2.1/3
- MIT License
- Mozilla Public License 2
- The Unlicense

But that's another issue I might open, so that we **really** get a choice rather than a (pre-filtered) subset.
Actually, IMHO, we should **always** get a wizard, where we developers select our requirements and GH then just finds (a) suitable license(s) for us.

Nonetheless, if you see **anything** I can add and/or improve to trigger the license badge mentioned above, please tell me. I'll do whatever and as fast as I can to get this finally resolved once and for all.

**Comment by mlinksva** [comment]
@Eagle3386 the tests in (1) are tests for this repo, have nothing to do with using any license in any other repo. (2) the files in this repo are line-wrapped at 78 characters so that they display nicely in a variety of contexts. The industry standard is that whitespace including line breaks are not material for license matching, in recognition that there's lots of variation from copying and pasting across formats etc, even for GPL. Check out https://github.com/spdx if you'd like to learn more.

Regarding your repo, great, it's EUPL-1.2 licensed. github.com doesn't display that fact, but it is nonetheless, just as any repo with a license file (which includes LICENSE or LICENSE.md) before github.com added a feature to display indicators for licenses it could detect.

There's nothing you can do to cause github.com to detect EUPL-1.2 right now, apologies if this was unclear from what I wrote above. There will not be until the API link mentioned above https://github.com/github/choosealicense.com/pull/523#issuecomment-355418428 returns info about EUPL-1.2 instead of "not found". My guess is this will be sometime after a non-beta version of licensee 9.9.0 is [released](https://github.com/benbalter/licensee/releases) because that's the version that first vendors EUPL-1.2, but that's just a guess not a promise. 😄 Thanks for your patience.

**Comment by Eagle3386** [comment]
@mlinksva Thanks for all the _valuable_  information regarding licenses and SPDX. Cleared much of the misunderstanding on my end. Apologies, if I sounded kinda "stubborn".

Since https://api.github.com/licenses/eupl-1.2 is not available yet, would it help specifying my code as EUPL-1.1 in order to receive the indicator (I assume that's how the information is called that I called "badge")?

**Comment by mlinksva** [comment]
@Eagle3386 if you value having an indicator/badge displayed above using the latest EUPL version, then yes you can switch to EUPL-1.1.

**Comment by Eagle3386** [comment]
While [one repo](https://github.com/Eagle3386/tzn-flake/blob/master/LICENSE.md) of mine is detected correctly as EUPL-1.2 now, [another one](https://github.com/Eagle3386/aya-platform/blob/master/LICENSE.md) is not - although both share the same file location (relatively to each repo's root), exactly matching each others content?

How can I fix that?

**Comment by mlinksva** [comment]
@Eagle3386 I don't know why they aren't both being detected; licensee from the command line detects both for me. You might write into http://github.com/c requesting re-detection, might be a glitch. Another option would be to change the filename to `LICENSE` which would trigger re-detection -- and avoid the terrible formatting that seems to result from treating the EUPL-1.2 text as markdown.

**Comment by Eagle3386** [comment]
> @Eagle3386 I don't know why they aren't both being detected; licensee from the command line detects both for me.

Glad, it actually worked - at least from the command line.

> You might write into [http://github.com/c](https://github.com/c) requesting re-detection, might be a glitch.

Did so just before commenting here today. Though, still no response up to now.

> Another option would be to change the filename to `LICENSE` which would trigger re-detection -- and avoid the terrible formatting that seems to result from treating the EUPL-1.2 text as markdown.

Thanks so much for that hint as it solved the issue! :+1:  :1st_place_medal:
After that, I even renamed it back to `LICENSE.md` - and it still shows `EUPL-1.2`.
Again, thanks a lot for your hint and support. Have a good one! :smile:

**Pull request merged**